SingularityUI/app/actions/api/base.es6

```diff
@@ -46,9 +46,20 @@ export function buildApiAction(actionName, opts={}) {
 
       let options = optsFunc(...args);
       return fetch(config.apiRoot + options.url, _.extend({credentials: 'include'}, _.omit(options, 'url')))
-        .then(response => response.json())
-        .then(json => {
-          dispatch(success(json));
+        .then(response => {
+          if (response.status >= 200 && response.status < 300) {
+            if (response.headers.get('Content-Type') === 'application/json') {
+              return response.json().then(json => dispatch(success(json)));
+            } else {
+              return response.text().then(body => dispatch(success({response: body})));
+            }
+          } else {
+            if (response.headers.get('Content-Type') === 'application/json') {
+              return response.json().then(body => dispatch(error(body)));
+            } else {
+              return response.text().then(body => dispatch(error({message: body})));
+            }
+          }
```

Comment:
you were getting the invalid JSON error because you were always trying to deserialize the JSON no matter what the `Content-Type` was. What i'm saying is update the promise chain to `fetch --> create appropriate object based on Content-Type value --> check HTTP status code, throw exception w/ appropriate data if not successful --> dispatch success action`, keeping the final `catch()` at the end to dispatch the error action